libs/core/src/lib/info-label/info-label.component.ts

```diff
@@ -12,79 +12,42 @@ type LabelType = 'numeric' | 'only-icon' | 'icon';
     changeDetection: ChangeDetectionStrategy.OnPush
 })
 export class InfoLabelComponent implements OnInit, OnChanges, CssClassBuilder {
-  
+    /** user's custom classes */
+    @Input()
+    class: string = '';
+
     /** 
      * The LabelType represented by the info label .
      * Can be one of the following: 'numeric' | 'only-icon' | 'icon'
      * For default info label omit this property
      */
-    _labelType: LabelType;
-    
     @Input()
-    set labelType(value: LabelType) {
-        this._labelType = value;
-        this.buildComponentCssClass();
-    }
-
-    get labelType(): LabelType {
-        return this._labelType;
-    }
-
-    /** define the icon type */
-    _glyph: string; 
+    type: LabelType;
 
+    /** glyph define the icon of info label */
     @Input()
-    set glyph(value: string) {
-        this._glyph = value;
-        this.buildComponentCssClass();
-    }
+    glyph: string;
 
-    get glyph(): string {
-        return this._glyph;
-    }
-
-    /**define the colour of the info label */
-    _color: string;
-    
+    /**define the colour of the info label starting form 1 to 10 */
     @Input()
-    set color(value: string) {
-        this._color = value;
-        this.buildComponentCssClass();
-    }
-
-    get color(): string {
-        return this._color;
-    }
-
-
-    _class: string = '';
-
-    @Input()
-    set class(value: string) {
-        this._class = value;
-        this.buildComponentCssClass();
-    }
-
-    get class(): string {
-        return this._class;
-    }
+    color: string;
 
     @applyCssClass
     buildComponentCssClass(): string {
```

Comment:
Can we try to avoid calling method from template? I know we are using it on several places, but its serious perf. problem. I would change the result out of this function call and assign only property to the template.  Angular needs to do allot of work, every time you click on that thing it needs to recalc. 

In this case I am not even sure we need to recalc. this every time. You initialize this only once and you are done. If you expect changes during lifecycle of the component, then you can recalc. this manually and save it. 

This is not issue only for this component but also for others. I am doing heavy caching and refactoring on my code as well. Thinking that OnPush solves the problem we are wrong. 

Angular introduced while ago something they called purePipes. 

Just imagine we will have 60 components like this and every components will do this call.

Reply:
I agree on that methods should not be used in templates. Getters behave a little bit different. I used them quite often. The biggest performance problem though is when there are too many flow controls added in template (like ngif, ngfor) even when You use properties on them the performance hit is huge (70 lines of ngif added to template  kills application) , thats why on push strategy comes in handy

Reply:
As per discussion betweek deno and frank this will be taken later, skipping this for now

Reply:
If everything works with with removing settings/getters and your stackblitz examples are running fine , then i am find with this. 

I dont think loki is going to refactor all this now, as this need to be solved once for all components.